engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/layout/text/TextUtilityTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2001 - 2018 Object Refinery Ltd, Hitachi Vantara and Contributors..  All rights reserved.
+ */
+
+package org.pentaho.reporting.engine.classic.core.layout.text;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.pentaho.reporting.libraries.fonts.awt.AWTFontRegistry;
+import org.pentaho.reporting.libraries.fonts.registry.FontMetricsFactory;
+import org.pentaho.reporting.libraries.fonts.registry.DefaultFontContext;
+import org.pentaho.reporting.libraries.fonts.registry.FontMetrics;
+import org.pentaho.reporting.libraries.fonts.registry.FontFamily;
+import org.pentaho.reporting.libraries.fonts.registry.FontRecord;
+
+
+public class TextUtilityTest {
+
+  @Test
```

Comment:
I would not put that test in here like that. You are testing against the runtime environment and you really cannot make any assumptions about the fonts they may have installed. I only added that 'non-test' in my example to demonstrate the fonts that may be bogus and to allow you to replicate these conclusions on your system. 

Your test will fail on a perfect system that has no bad fonts.

However, what you can test the function more reasonably by passing a mock of a fontmetrics object in and validate the outcome both both cases (one for a good font, one for a bad font).

As the Excel output target does not allow us to inject our own debug-level font contex into the process, there is no safe way to isolate the test from the runtime (so that we dont rely on fonts produced by the native AWT) so there is no safe way to create integration tests that aren't bogus or prone to failure. So the excel export for this case must be validated manually by QA later.